resolver/resolver.go

```diff
@@ -349,8 +349,8 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
 		}
 	}
 
-	err = w.WriteMsg(m)
-	if err != nil {
+	m.Compress = true
```

Comment:
would like to see some TODO comments here describing the pain point and the potential problems that a user will see until we resolve this properly. Perhaps even a reference to tickets 173 and 174.